api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,18 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  try {
+    const accessToken = req.headers['x-access-token'];
+    if (!accessToken) {
+      throw new ClientError(401, 'authentication required');
+    }
+    const payload = jwt.verify(accessToken, process.env.TOKEN_SECRET);
+    req.user = payload;
+    next();
+  } catch (err) {
+    next(err);
+  }
```

Comment:
while this does work, it is not needed to route the errors to catch since the errors are already being handled